WiserTaskScheduler/WiserTaskScheduler/Modules/ServerMonitors/Services/ServerMonitorsService.cs

```diff
@@ -0,0 +1,443 @@
+
+            //Check if the ram is above the threshold.
+            if (ramValue > threshold)
+            {
+                emailRamSent = false;
+                return;
+            }
+
+            //Check if an email already has been sent.
+            if (emailRamSent)
+            {
+                return;
+            }
+
+            emailRamSent = true;
+            await gclCommunicationsService.SendEmailAsync(receiver, subject, body);
+        }
+
+        /// <summary>
+        /// Checks the Cpu usage detection type to use and calls the right function.
+        /// </summary>
+        /// <param name="monitorItem">The information for the monitor action.</param>
+        /// <param name="threshold">The threshold to check if an email needs to be send.</param>
+        /// <param name="gclCommunicationsService">The name of the service in the configuration, used for logging.</param>
+        /// <param name="configurationServiceName">The communications service from the GCL to actually send out the email.</param>
+        /// <returns></returns>
+        private async Task GetCpuUsageAsync(ServerMonitorModel monitorItem, int threshold, string configurationServiceName, CommunicationsService gclCommunicationsService)
+        {
+            //gets the detection type of cpu usage to use.
+            switch(monitorItem.CpuUsageDetectionType)
+            {
+                case CpuUsageDetectionTypes.ArrayCount:
+                    await GetCpuUsageArrayCountAsync(monitorItem, threshold, configurationServiceName, gclCommunicationsService);
+                    break;
+                case CpuUsageDetectionTypes.Counter:
+                    await GetCpuUsageCounterAsync(monitorItem, threshold, configurationServiceName, gclCommunicationsService);
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Gets the CPU usage with the use of a counter which checks if for a certain amount of times the value above the threshold is.
+        /// </summary>
+        /// <param name="monitorItem">The information for the monitor action.</param>
+        /// <param name="threshold">The threshold to check if an email needs to be send.</param>
+        /// <param name="gclCommunicationsService">The name of the service in the configuration, used for logging.</param>
+        /// <param name="configurationServiceName">The communications service from the GCL to actually send out the email.</param>
+        /// <returns></returns>
+        private async Task GetCpuUsageArrayCountAsync(ServerMonitorModel monitorItem, int threshold, string configurationServiceName, CommunicationsService gclCommunicationsService)
+        {
+            //the first value of performance counter will always be 0.
+            if (!firstValueUsed)
+            {
+                firstValueUsed = true;
+                var firstValue = cpuCounter.NextValue();
```

Comment:
Variabele vervalt meteen door eindigen scope dus het resultaat uit de functie hoeft niet in een variabele te komen.